src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -44,28 +47,52 @@ public final void publishWithDelay(Message message, long delayMilliseconds) thro
             log.warn("Publishing delayed message to non-delayed queue queue:{}", queueName);
         }
 
+        val properties = new AMQP.BasicProperties.Builder();
+
+        if (config.getCompressionConfig() != null &&
+                config.getCompressionConfig().isEnableCompression() &&
+                config.getCompressionConfig().getCompressionAlgorithm() != null) {
+            properties.headers(ImmutableMap.of("X-CompressionType",
+                    config.getCompressionConfig().getCompressionAlgorithm()));
+        }
+
+
         if (config.getDelayType() == DelayType.TTL) {
             publishChannel.basicPublish(ttlExchange(config),
                     queueName,
-                    new AMQP.BasicProperties.Builder()
-                            .expiration(String.valueOf(delayMilliseconds))
+                    properties.expiration(String.valueOf(delayMilliseconds))
                             .deliveryMode(2)
                             .build(),
                     mapper().writeValueAsBytes(message));
         } else {
-            publish(message, new AMQP.BasicProperties.Builder()
-                    .headers(Collections.singletonMap("x-delay", delayMilliseconds))
+            publish(message, properties.headers(Collections.singletonMap("x-delay", delayMilliseconds))
                     .deliveryMode(2)
                     .build());
         }
     }
 
     public final void publish(Message message) throws Exception {
+        if (config.getCompressionConfig() != null && config.getCompressionConfig().isEnableCompression()) {
```

Comment:
As discussed here, pass it through a series of decorators/transformers that will add headers etc and use the output of the final decorator to call publish